include/ttmlir/Target/TTNN/TTNNToFlatbuffer.h

```diff
@@ -0,0 +1,24 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef TTMLIR_TARGET_UTILS_TTNNTOFLATBUFFER_H
+#define TTMLIR_TARGET_UTILS_TTNNTOFLATBUFFER_H
+
+#include "mlir/IR/Operation.h"
+#include "mlir/Support/LogicalResult.h"
+
+namespace mlir::tt::ttnn {
+
+// Convert a TTNNIR operation to a flatbuffer
+// As a convenience, this function returns the flatbuffer as a vector of bytes
+// since we need to know the size of the buffer
+std::vector<uint8_t> ttnnToFlatbuffer(Operation *op);
```

Comment:
`std::vector<uint8_t>` might be awkward for sharing with the runtime, now you'll have to manually copy and wrap into a shared_ptr, whereas previously it could just be passed directly over to runtime.

Maybe this could be a pair? `std::pair<std::shared_ptr<void>, std::size_t>`, could prevent unnecessary copying of the `std::vector<uint8_t>` too in case it gets big with embedded tensor data.

Reply:
Another option, is that the flatbuffer is prefixed with the size, we could also pull it from there (this is how runtime does it):
```
auto size = ::flatbuffers::GetSizePrefixedBufferLength(
    static_cast<const uint8_t *>(handle.get()));
```

Reply:
I wanted to avoid pulling flatbuffer headers into tt-forge. Its fine to use pair I guess.

Reply:
In tt-forge we don't include flatbuffer headers to read its size, it is all handled by included runtime. @pilkicTT, correct me if I am wrong here?

Reply:
We don't include them directly, we include `runtime/types.h` which will include it later on.

However, I don't think we'll need the size and I think we cannot avoid including these headers, since we will need to interact with the flatbuffer in tt-forge. So, returning the `shared_ptr<void>` looks good to me.

Reply:
Done